src/frontend/src/optimizer/property/monotonicity.rs

```diff
@@ -257,7 +274,13 @@ impl MonotonicityAnalyzer {
                             .as_literal()
                             .and_then(|literal| literal.get_data().as_ref())
                             .map(|tz| tz.as_utf8().as_ref());
-                        if time_zone_is_without_dst(time_zone) {
+                        let date_unit = func_call.inputs()[0]
+                            .as_literal()
+                            .and_then(|literal| literal.get_data().as_ref())
+                            .map(|tz| tz.as_utf8().as_ref());
+                        if time_zone_is_without_dst(time_zone)
+                            || date_unit_larger_than_day(date_unit)
+                        {
```

Comment:
Can you please add an example to demonstrate the correctness?

What I can think of is like in a timezone with DST, `MM-dd 23:50:00` + interval `12 minutes` may be `MM-dd 23:02:00`, while the `date_trunc('day', ...)` result be `MM-dd 00:12:00` versus `MM-dd (00:00:00)`. Seems this will break the correctness. Am I understanding it wrong?

Reply:
https://arc.net/l/quote/wfxyyukb